PyApps/src/Plugins/display_image_qt5.py

```diff
@@ -806,9 +845,17 @@ def update_spectrum_display(self, menuid):
         self.source_marker = None
         self.is_combined_image = False
 
-# if we got here, emit signal up to result_plotter here
-      menuid = self.spectrum_menu_items - 1 - menuid
-      self.emit(Qt.SIGNAL("handle_spectrum_menu_id"),(menuid,))
+# if we got here, emit signal up to result_plotter 
+# NOTE: for python3/Qwt6 we see to get here in ways I do NOT understand
+# and I handle this with an exception. Maybe some day I'll figure this out :) 
+      try:
+# menuid is a python object in python3  so how can I be doing 
+# arithmetic on it? # how did this ever work in python2?
+        menuid = self.spectrum_menu_items - 1 - menuid
+        self.handle_spectrum_menu_id.emit(menuid)
+      except:
```

Comment:
Qualify exception to be caught. This catches all exceptions and returns no message? Looks quite dangerous

Reply:
Still working on this ...